### DISCUSSION TITLE
annovarToMaf particular cases of mixed annotations

### DISCUSSION THREAD

**Discussion by lisemangiante** [PoisonAlien/maftools#680] [General]
Hi PoisonAlien,
First, thank you for this remarkable tool! I’m using it to get meaningful oncoplots from our annovar outputs. In the practice of our data, I noticed a few points in the annovarToMaf code that I would like to discuss. 

1. In lines 84-85, you mentioned that in case of having “X1;X2” in the Gene.refGene annotation column, you choose to keep only X1 since this is the closest gene from the variant position I suppose. However, in the case of “exonic” variants (Func.refGene), I don’t see why this argument is relevant, since, from what I understood from annovar documentation, the variant is exonic for both gene, so the variant can be considered as equally interesting for both genes, right?

2. In lines 120+130, ann_exonic object is set for “exonic” variants (Func.refGene), right? I don’t see how “exonic;splicing” variants could be integrated in this object. Beside, in our data, all “exonic;splicing” variants (all non-synonymous) from annovar are converted into “RNA” Variant_Classification by annovar_values and not "Missense_Mutation" as we could expect from lines 130-133.  Therefore, I guess “exonic;splicing” variants went within the ann_res object (line 165).

3. Finally, in line 168, ann_res object can contain “ncRNA;splicing” for instance, and in this case only the “ncRNA” annotation is kept for these variants. However, if two genes are given “X1;X2” in the Gene.refGene column, it means that X1 is ncRNA and X2 is splicing, from what I understood from annovar documentations. Splicing events is one of the items that you chose to integrate into the oncoplot by default. Here, I have the feeling that by doing that we might miss some variants that we would like to integrate within the final oncoplot.

Thank you again for your help,
Best,
Lise

**Comment by PoisonAlien** [comment]
Hello,
Thanks for the well-researched post. All your points are valid but unfortunately, there is no straightforward answer. 

1. This is the issue of `transcript prioritization`. It is hard to guess which transcript a given mutation is affecting. Annovar seems to assign them to all possible affected transcripts and organizes them by closeness. I would suggest using VEP or better [vcf2maf](https://github.com/mskcc/vcf2maf) which will perform transcript prioritization for you.
2. Here, in `exonic;splicing` the first entry of `exonic` is chosen. `exonic` variants are not necessarily non-synonymous. The definition is still ambiguous to me as well. But for what it's worth, annovar `exonic` terms are `exon_variant` in [SO terms](http://www.sequenceontology.org/browser/current_release/term/SO:0001791) that can arise from three possible sources (see the link for details). VEP [classifies](https://github.com/mskcc/vcf2maf/blob/main/vcf2maf.pl#L993) them as RNA, and it's the same here.
3. This is the same as your first point and concerns the `transcript prioritization` - whether to assign this variant to `ncRNA` or `splice site`.  

I am also attaching a file here for `annovar to VEP to MAF` mappings, in case if you would like to check it. 
I hope this helps. 

[annovar_to_MAF_mappings.txt](https://github.com/PoisonAlien/maftools/files/6107433/annovar_to_MAF_mappings.txt)
